src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -31,6 +33,16 @@ public static void SeekReaderForwardToMatchByteSignature(BinaryReader reader, by
                 throw new ArgumentNullException(nameof(reader));
             }
 
+            if (byteSignature == null)
+            {
+                throw new ArgumentNullException(nameof(byteSignature));
+            }
+
+            if (byteSignature.Length == 0)
+            {
+                throw new InvalidDataException(nameof(byteSignature));
```

Comment:
`ArgumentException`

You could make a new resource string (`ArgumentCannotBeNullOrEmpty`):

    The argument cannot be null or empty.

...and then combine these two tests into:

    if (byteSignature == null || byteSignature.Length == 0)
    {
        throw new ArgumentException(Strings.ArgumentCannotBeNullOrEmpty, nameof(byteSignature));
    }

...and then update all the other places in this PR to do the same.